bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/ButtonVirtualDatapointHandler.java

```diff
@@ -57,33 +61,64 @@ public boolean canHandleEvent(HmDatapoint dp) {
 
     @Override
     public void handleEvent(VirtualGateway gateway, HmDatapoint dp) {
```

Comment:
You have made major changes to the logic here. Is the behavior regarding the triggered events really like before? It would be bad if rules behaved differently than before, because the recognition of a double click or long press works differently.

I don't have any buttons in use right now, otherwise I could do some regression tests.

Reply:
I hope so ;-) The goal is SINGLE_PRESS being triggered as before (just with shorter spacing due to the changed DOUBLE_PRESS handling), LONG_PRESS being triggered as before and just the new 2 events being triggered in addition to that.

Reply:
With the removal of DOUBLE_PRESSED events, the goal is now
- triggering SHORT_PRESSED events on *every* short button press (before it was either SHORT_PRESSED or DOUBLE_PRESSED, depending on timing)
- triggering LONG_PRESSED events on every long press start (as before)
- triggering LONG_REPEATED events on long press repetition (CONT being sent from device; new, didn't exist before)
- triggering LONG_RELEASED events on long press end (new, didn't exist at all before)